core/include/detray/utils/grid/detail/axis_binning.hpp

```diff
@@ -167,6 +167,16 @@ struct regular {
 
         return {min, max};
     }
+
+    /// Equality operator
+    ///
+    /// @param rhs the axis to compare to
+    ///
+    /// @returns whether the two axes are equal
+    DETRAY_HOST_DEVICE constexpr bool operator==(const regular &rhs) const {
+        return m_offset == rhs.m_offset && m_n_bins == rhs.m_n_bins &&
+               m_bin_edges == rhs.m_bin_edges;
```

Comment:
This is comparing the bin_edges pointers, right? Can binning that points to different bin edges locations still be the same if the edge values are the same?

Reply:
I would suggest to following:

```c++
m_bin_edges == rhs.m_bin_edges || bin_edges() == rhs.bin_edges()
```